src/Configuration/src/CloudFoundry/JsonStreamConfigurationProvider.cs

```diff
@@ -11,7 +11,7 @@ internal sealed class JsonStreamConfigurationProvider : JsonConfigurationProvide
 {
     private readonly JsonStreamConfigurationSource _source;
 
-    internal JsonStreamConfigurationProvider(JsonStreamConfigurationSource source)
+    public JsonStreamConfigurationProvider(JsonStreamConfigurationSource source)
```

Comment:
It doesn't change the public API surface, because the containing type is internal.

Some types that used to be public were later made internal. Then the question arises of how to declare nested members: internal+public or internal+internal? Technically it doesn't matter, but I've been using the following convention:
- internal+internal: In case this type ever becomes public, the member needs to stay internal. For example, we sometimes internally expose the ILogger so tests can access it, but it should never be exposed publicly.
- internal+public: In case this type ever becomes public, it makes sense that the member becomes public as well, so the type can be used effectively. Because we use `[InternalsVisibleTo]` in tests, such use cases don't always surface as a compile error.

In this specific case: if the ConfigurationProvider ever becomes public, so should its associated ConfigurationSource. Now if someone forgets to do that, a build error will show up, along the lines of "ConfigurationSource is less visible", suggesting to make it public as well. Keeping the constructor internal would not produce that error.

Based on the above, I've pushed a commit to tweak these a bit more because I didn't consistently apply this convention.